### PR TITLE
chore: CHARTS-6802 Add github action to check markdown links

### PR DESCRIPTION
**Pull request by khanguslee**
Added a reusable github action workflow to check if any links in the repository are valid. 

I have set the workflow to run on every push/PR and every week.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Main workflow
+
+on:
+  push:
+    branches:
+      - '**'
```
Comment (shaketbaby) [inline .github/workflows/main.yml:6]: Probably `master` or `main` branches only on push?

File: .github/workflows/markdown-link-check.config.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "ignorePatterns": [
+    {
+      "pattern": "^http://localhost.*"
```
Comment (mongo-J) [inline .github/workflows/markdown-link-check.config.json:4]: Shall we also include `https://localhost` for future proof
```suggestion
      "pattern": "^https?://localhost.*"
```

File: .github/workflows/markdown-link-check.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  check-links:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        use-verbose-mode: 'yes'
+        config-file: '.github/workflows/markdown-link-check.config.json'
+        folder-path: './examples'
```
Comment (mongo-J) [inline .github/workflows/markdown-link-check.yml:15]: Do we want to limit to just the `folder-path` or we let the plugin to check all markdown files.

Comment (khanguslee) [reply .github/workflows/markdown-link-check.yml:15]: I think it's okay to check all markdown files. I feel like it'll be a lot more work to add every folder path for every example. We also want to check _every_ link that is public to make sure that it isn't broken.

**Comment by mongo-J** [comment]
Also, just wondering have we tested this action with an actual broken link and seeing the pipeline fail, I can't seem to see it from the previous running actions.

**Comment by khanguslee** [comment]
> Also, just wondering have we tested this action with an actual broken link and seeing the pipeline fail, I can't seem to see it from the previous running actions.

See https://github.com/mongodb-js/charts-embed-sdk/actions/runs/3034642385/jobs/4883941010 where localhost was failing and why I needed to add the config.

File: .github/workflows/markdown-link-check.yml (hunk 3 above)
Comment (mongo-J) [reply .github/workflows/markdown-link-check.yml:15]: If that's what we want, then shall we remove this line `folder-path` as reading here, that is not default behaviour unless `folder-path` is defined.

Comment (khanguslee) [reply .github/workflows/markdown-link-check.yml]: Sounds good. Removed `file-path` too so that we check every single `.md` file.

**Review by mongo-J — APPROVED** [review]
LGTM 👍

**Pull request merged**